### ISSUE TITLE
Error page needs to have an easily copyable exception message and stack trace

### ISSUE THREAD

**Issue by kichalla** [aspnet/Diagnostics#33]
Similar to #12 

Currently when an unhandled exception occurs, following is the html output that we see...as you can imagine this makes it difficult to copy the stack trace and send it out to someone to take a look....

```html
    <body>
        <h1>An unhandled exception occurred while processing the request.</h1>
                <h2 class="titleerror">OutOfMemoryException: Exception of type &#39;System.OutOfMemoryException&#39; was thrown.</h2>
                    <p class="location">System.Data.SqlClient.TdsParser.ProcessSSPI(Int32 receivedLength) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\TdsParser.cs">TdsParser.cs</code>, line 5751</p>
                <h2 class="titleerror">DataStoreException: An error occured while running a data store operation. See InnerException for details.</h2>
                    <p class="location">Microsoft.Data.Entity.Query.EntityQueryExecutor.EnumerableExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()</p>
        <ul id="header">
                <li id="stack" tabindex="1" class="selected">
                    Stack
                </li>
                            <li id="query" tabindex="2">
                    Query
                </li>
                            <li id="cookies" tabindex="3">
                    Cookies
                </li>
                            <li id="headers" tabindex="4">
                    Headers
                </li>
                            <li id="environment" tabindex="5">
                    Environment
                </li>
        </ul>
            <div id="stackpage" class="page">
                <ul>

                        <li>
                            <h2 class="stackerror">OutOfMemoryException: Exception of type &#39;System.OutOfMemoryException&#39; was thrown.</h2>
                            <ul>
                                <li class="frame" tabindex="6">

                                        <h3>System.Data.SqlClient.TdsParser.ProcessSSPI(Int32 receivedLength) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\TdsParser.cs">TdsParser.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="7">

                                        <h3>System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean&amp; dataReady) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\TdsParser.cs">TdsParser.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="8">

                                        <h3>System.Data.SqlClient.TdsParser.Run(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\TdsParser.cs">TdsParser.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="9">

                                        <h3>System.Data.SqlClient.SqlInternalConnectionTds.CompleteLogin(Boolean enlistOK) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlInternalConnectionTds.cs">SqlInternalConnectionTds.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="10">

                                        <h3>System.Data.SqlClient.SqlInternalConnectionTds.AttemptOneLogin(ServerInfo serverInfo, String newPassword, String newSecurePassword, Boolean ignoreSniOpenTimeout, TimeoutTimer timeout, Boolean withFailover) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlInternalConnectionTds.cs">SqlInternalConnectionTds.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="11">

                                        <h3>System.Data.SqlClient.SqlInternalConnectionTds.LoginNoFailover(ServerInfo serverInfo, String newPassword, String newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString connectionOptions, SqlCredential credential, TimeoutTimer timeout) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlInternalConnectionTds.cs">SqlInternalConnectionTds.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="12">

                                        <h3>System.Data.SqlClient.SqlInternalConnectionTds.OpenLoginEnlist(TimeoutTimer timeout, SqlConnectionString connectionOptions, SqlCredential credential, String newPassword, String newSecurePassword, Boolean redirectedUserInstance) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlInternalConnectionTds.cs">SqlInternalConnectionTds.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="13">

                                        <h3>System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, String newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlInternalConnectionTds.cs">SqlInternalConnectionTds.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="14">

                                        <h3>System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlConnectionFactory.cs">SqlConnectionFactory.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="15">

                                        <h3>System.Data.ProviderBase.DbConnectionFactory.CreateNonPooledConnection(DbConnection owningConnection, DbConnectionPoolGroup poolGroup, DbConnectionOptions userOptions) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\ProviderBase\DbConnectionFactory.cs">DbConnectionFactory.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="16">

                                        <h3>System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal&amp; connection) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\ProviderBase\DbConnectionFactory.cs">DbConnectionFactory.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="17">

                                        <h3>System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\ProviderBase\DbConnectionClosed.cs">DbConnectionClosed.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="18">

                                        <h3>System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry) in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlConnection.cs">SqlConnection.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="19">

                                        <h3>System.Data.SqlClient.SqlConnection.Open() in <code title="C:\BuildAgent\work\7894bcffbd50f0b2\src\System.Data.SqlClient\System\Data\SqlClient\SqlConnection.cs">SqlConnection.cs</code></h3>

                                </li>
                                <li class="frame" tabindex="20">

                                        <h3>Microsoft.Data.Entity.Relational.RelationalConnection.Open()</h3>

                                </li>
                                <li class="frame" tabindex="21">

                                        <h3>Microsoft.Data.Entity.Relational.Query.QueryMethodProvider.Enumerable`1.Enumerator.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="22">

                                        <h3>System.Linq.Enumerable.&lt;SelectManyIterator&gt;d__14`2.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="23">

                                        <h3>System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="24">

                                        <h3>Microsoft.Data.Entity.Query.EntityQueryExecutor.EnumerableExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()</h3>

                                </li>
                            </ul>
                        </li>
                        <li>
                            <h2 class="stackerror">DataStoreException: An error occured while running a data store operation. See InnerException for details.</h2>
                            <ul>
                                <li class="frame" tabindex="25">

                                        <h3>Microsoft.Data.Entity.Query.EntityQueryExecutor.EnumerableExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="26">

                                        <h3>System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)</h3>

                                </li>
                                <li class="frame" tabindex="27">

                                        <h3>System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)</h3>

                                </li>
                                <li class="frame" tabindex="28">

                                        <h3>Microsoft.AspNet.Identity.EntityFramework.UserStore`4.GetClaimsAsync(TUser user, CancellationToken cancellationToken)</h3>

                                </li>
                                <li class="frame" tabindex="29">

                                        <h3>Microsoft.AspNet.Identity.UserManager`1.&lt;GetClaimsAsync&gt;d__1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="30">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="31">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="32">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="33">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="34">

                                        <h3>Microsoft.AspNet.Identity.ClaimsIdentityFactory`2.&lt;CreateAsync&gt;d__1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="35">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="36">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="37">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="38">

                                        <h3>Microsoft.AspNet.Identity.SignInManager`1.&lt;CreateUserIdentityAsync&gt;d__1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="39">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="40">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="41">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="42">

                                        <h3>Microsoft.AspNet.Identity.SignInManager`1.&lt;SignInAsync&gt;d__1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="43">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="44">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="45">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="46">

                                        <h3>Microsoft.AspNet.Identity.SignInManager`1.&lt;SignInOrTwoFactorAsync&gt;d__1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="47">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="48">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="49">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="50">

                                        <h3>Microsoft.AspNet.Identity.SignInManager`1.&lt;PasswordSignInAsync&gt;d__1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="51">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="52">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="53">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="54">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()</h3>

                                </li>
                                <li class="frame" tabindex="55">

                                        <h3>MusicStore.Controllers.AccountController.&lt;Login&gt;d__2.MoveNext() in <code title="C:\StressTestingApps\3f69ca46422246978dc5e6941ccf0caf\3f69ca46422246978dc5e6941ccf0caf\Controllers\AccountController.cs">AccountController.cs</code></h3>

                                        <div class="source">
                                                <ol start="34" class="collapsible">
                                                        <li><span>        [AllowAnonymous]</span></li>
                                                        <li><span>        [ValidateAntiForgeryToken]</span></li>
                                                        <li><span>        public async Task&lt;IActionResult&gt; Login(LoginViewModel model, string returnUrl = null)</span></li>
                                                        <li><span>        {</span></li>
                                                        <li><span>            if (ModelState.IsValid)</span></li>
                                                        <li><span>            {</span></li>
                                                </ol>

                                            <ol start="40" class="highlight">
                                                <li><span>                var signInStatus = await SignInManager.PasswordSignInAsync(model.UserName, model.Password, model.RememberMe, shouldLockout: false);</span></li></ol>

                                                <ol start='41' class="collapsible">
                                                        <li><span>                switch (signInStatus)</span></li>
                                                        <li><span>                {</span></li>
                                                        <li><span>                    case SignInStatus.Success:</span></li>
                                                        <li><span>                        return RedirectToLocal(returnUrl);</span></li>
                                                        <li><span>                    case SignInStatus.LockedOut:</span></li>
                                                        <li><span>                        ModelState.AddModelError(&quot;&quot;, &quot;User is locked out, try again later.&quot;);</span></li>
                                                </ol>
                                        </div>
                                </li>
                                <li class="frame" tabindex="56">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="57">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="58">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)</h3>

                                </li>
                                <li class="frame" tabindex="59">

                                        <h3>Microsoft.AspNet.Mvc.ReflectedActionExecutor.&lt;CastToObject&gt;d__2`1.MoveNext()</h3>

                                </li>
                                <li class="frame" tabindex="60">

                                        <h3>--- End of stack trace from previous location where exception was thrown ---</h3>

                                </li>
                                <li class="frame" tabindex="61">

                                        <h3>System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)</h3>

                                </li>
                                <li clas
```

**Comment by Eilon** [comment]
See https://github.com/aspnet/Diagnostics/issues/60 for some ideas on how to implement this.

**Comment by ryanbrandenburg** [comment]
Fixed in #189
